### PR TITLE
Table services

### PR DESCRIPTION
**Pull request by Mih4lych**
- Tables were separated.
- Added card table service.
- Changed CardProcess with using new service.
- Docker config and DB DDl/config.

### CODE REVIEW

File: db/init-scripts.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+create table if not exists "Card" ("id" VARCHAR NOT NULL PRIMARY KEY,"gameId" VARCHAR NOT NULL,"word" VARCHAR NOT NULL,"cardRole" VARCHAR NOT NULL,"cardState" VARCHAR NOT NULL);
```
Comment (nasta-k) [inline db/init-scripts.sql:1]: Foreign keys should be configured on migration level

File: src/main/scala/domain/GameError.scala [hunk 2]
```diff
@@ -15,4 +18,10 @@ object GameError extends Enum[GameError] {
   case object SpymasterAlreadyInTeamError extends GameError
   case object TeamWithoutOperativesError extends GameError
   case object TeamWithoutSpymasterError extends GameError
+
+  case object CardNotFoundError extends GameError
```
Comment (nasta-k) [inline src/main/scala/domain/GameError.scala:22]: Could be generalised as case class

File: src/main/scala/game/CardProcess.scala [hunk 3]
```diff
@@ -22,25 +28,33 @@ object CardProcess {
...
           startingTeam match {
-            case TeamColor.Blue => Vector(CardRole.Agent(TeamColor.Red))
-            case TeamColor.Red => Vector(CardRole.Agent(TeamColor.Blue))
+            case TeamColor.Blue => List(CardRole.Agent(TeamColor.Red))
+            case TeamColor.Red  => List(CardRole.Agent(TeamColor.Blue))
           }
         }
       }
+
+      override def playCard(cardId: CardId): ErrorOrT[F, Unit] = {
```
Comment (nasta-k) [inline src/main/scala/game/CardProcess.scala:53]: Shouldn't it return `Card` instance?

File: src/main/scala/service/db/CardTableService.scala [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.Transactor
+
+trait CardTableService[F[_]] {
+  def getAllCardsByGameId(gameId: GameId): F[Seq[Card]]
+  def getCardById(id: CardId): F[Option[Card]]
+  def insert(cards: Seq[Card]): F[Unit]
+  def update(card: Card): F[Unit]
+}
+object CardTableService {
```
Comment (nasta-k) [inline src/main/scala/service/db/CardTableService.scala:16]:
```suggestion
object CardTableDbService {
```
Not quite self explanatory, that service covers db layer operation

File: src/main/scala/service/db/CardTableService.scala [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  def apply[F[_]: Async](transactor: Resource[F, Transactor[F]]): CardTableService[F] =
```
Comment (nasta-k) [inline src/main/scala/service/db/CardTableService.scala:17]: Should not be referenced as `Resource` in services

Create as resource ( probably would be in `Main` runnable object) and use `Transactor[F]`

File: src/main/scala/domain/GameError.scala (hunk 2 above)
Comment (Mih4lych) [reply src/main/scala/domain/GameError.scala:22]: You mean something like `case class CardNotFoundError(reason: String)` or what?

File: src/main/scala/game/CardProcess.scala (hunk 3 above)
Comment (Mih4lych) [reply src/main/scala/game/CardProcess.scala:53]: Right now I'm trying to generalize my logic with Unit as a result. With events I'll change it.

File: src/main/scala/domain/GameError.scala (hunk 2 above)
Comment (nasta-k) [reply src/main/scala/domain/GameError.scala:22]: `case class NotFoundError(entity: String)` or even id might be included

**Pull request merged**